- Primera version de captura de tweets asíncrona

alejandrozf opened the pull request:
- Creo que ya estarían siendo asíncronas las peticiones

Commit:
- Primera version de captura de tweets asíncrona

Commit:
- Completando bajada asincrona cambiando twisted por gevent

pablocelayes commented:
@alejandrozf encontraste qué es lo que te fallaba de la autenticación de la API?
Prefiero probarlo cuando tengas resuelto eso.

alejandrozf commented:
Yo había detectado un caracter q faltaba en un token, se lo adicione y me
andaba ya sin errores localmente, y actualice la planilla, a ti todavía te
falla??
Si es así si puedes pasame los tokens q te fallan para ver como lo soluciono
El jul 19, 2017 13:51, "Pablo Celayes" <notifications@github.com> escribió:

> @alejandrozf <https://github.com/alejandrozf> encontraste qué es lo que
> te fallaba de la autenticación de la API?
> Prefiero probarlo cuando tengas resuelto eso.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/pablocelayes/paso2017sna/pull/1#issuecomment-316448661>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AH_VJI_jiTmcW36M6A3K-Ntlniqnge2pks5sPjQpgaJpZM4OY2SI>
> .
>

pablocelayes commented:
No, no me falla. Simplemente no lo probé porque me quedé con lo que decía tu comentario anterior.
Te aviso entonces cuando lo pruebe, pero si te anda, debería andar por acá.

Commit:
- Modificando API para cerrar en una sola llamada las conexiones para un
  id de usuario ya con todos los tweets descargados

Commit:
******* Iniciando trabajo en pruebas de las versiones sincronas y
        asincronas de la bajada de tweets
- Adaptado codigo sincrono de la notebook '0.4-Traer tweets Llaryora Baldassi'

Commit:
- Armado script de bajada de tweets sincronica a partir de notebook provista
- Calculado tiempo promedio de tweets, de tweets por usuario, y tiempo total

Commit:
- Imprimiendo en pantalla las estadisticas de los tweets descargados

Commit:
- Cambios menores

Commit:
- Fix en API síncrona

Commit:
- Cambiando la llamada de traer timeline

Commit:
- Roll back a version original síncrona
- Adicionando indicativo 'segundos' a los reportes de tiempo

Commit:
- Optimizando uso de las conexiones en version asìncrona
- Reduciendo mensajes informativos
- Refactor de código de captura de tweets
- Cambiando a 'defaultdict' para almacenar datos de tweets y conexiones
por usuario

Commit:
- Completando ambos reportes

Commit:
- Adicionando posibilidad de descargar N páginas de tweets para ambas
  versiones (sincrónica y asíncrona), marcando la nueva opción en los
  scripts de descarga

Commit:
- Adicionando backend de mongodb a API sincrónica, salvando tweets por
  página una vez son descargados

Commit:
- Fix en uso de gevent, ahora las conexiones son realmente asíncronas
- Integrando con backed en MongoDB
- Volviendo el script de descarga a la version de rangos de fechas
- Fix en la forma de calcular el tiempo promedio de descarga de todos los tweets
de un usuario

Commit:
- Unificando manera de salvar los tweets en la BD

Commit:
- Usando pool para limitar conexiones concurrentes

Commit:
- Evitando duplicados en la BD

Commit:
- Cambios menores

Commit:
- Eliminando código deprecado

Commit:
- Agregando utilidades para manipular la BD con Pymongo

Commit:
- Cambiando esquema asíncrono usando POOL y creando conexiones por uid
  para reducir los tiempos por la explosión combinatoria de greenlets de generar
  conexiones por paginas
- Rotando asíncronamente las conexiones convenientemente
- Validando el caso de que no hubiese tweets descargados en las
  estadisticas finales

Commit:
- Eliminando variables nos usadas
- Actualizando rango de días a descargar

Commit:
- Fix al concatenar archivos .json

Commit:
- Fix en el chequeo de duplicados

Commit:
- Mejor definicion y manejo del control de las páginas descargadas por usuario
